nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/state/server/ZooKeeperStateServer.java

```diff
@@ -104,8 +105,16 @@ private void startStandalone() throws IOException {
             embeddedZkServer.setMinSessionTimeout(config.getMinSessionTimeout());
             embeddedZkServer.setMaxSessionTimeout(config.getMaxSessionTimeout());
 
+            final InetSocketAddress port = config.getClientPortAddress();
+            final InetSocketAddress secPort = config.getSecureClientPortAddress();
             connectionFactory = ServerCnxnFactory.createFactory();
-            connectionFactory.configure(config.getClientPortAddress(), config.getMaxClientCnxns());
+            if (secPort != null) {
```

Comment:
Since the insecure client port and secure client port aren't mutually exclusive you should add the checks for each port independently and setup two connection factories with their own startup and shutdown.

Reply:
I think you're right, your implementation should be merged, not this one.

I was trying to avoid multiple connections due to the security implications but I think we can just as easily handle that elsewhere.